documentation/src/docs/wire-protocol.md

```diff
@@ -7,106 +7,103 @@ You may obtain a copy of the License at
 
-1.  Data
 
-### Data Appended - Reply
+## Data Appended - Reply
 
-1.  ConnectionId - UUID (16 bytes)
-2.  ConnectionOffsetAckLevel - Long (8 bytes) The highest offset before which all data is successfully stored on this segment for this connectionId
+1. `writerId`: UUID (16 bytes).It identifies the requesting appender.
+2. `eventNumber`: Long (8 bytes). This matches the `lastEventNumber` in the append block.
+3. `previousEventNumber`: Long (8 bytes). This is the previous value of `eventNumber` that was returned in the last `DataAppeneded`.
 
-When appending a client
 
-1.  Establishes a connection to what it thinks is the correct host.
-2.  Sends a Setup Append request.
-3.  Waits for the Append Setup reply.
+When appending a client:
 
-Then it can
-1.  Send a BeginEventBlock request
-2.  Send as many messages as can fit in the block
-3.  Send an EndEventBlock request
+- Establishes a connection to the host chosen by it.
+- Sends a "Setup Append" request.
+- Waits for the "Append Setup" reply.
 
-While this is happening the server will be periodically sending it DataAppended replies acking messages. Note that there can be multiple "Appends" setup for a given TCP connection. This allows a client to share a connection when producing to multiple segments.
+After receiving the "Append Setup" reply, it performs the following:
+- Send a `AppendBlock` request.
+- Send as many Events that can fit in the block.
+- Send an `AppendBlockEnd` request.
 
-A client can optimize its appending by specifying a large value in it's BeginAppendBlock message, as the events inside of the block do not need to be processed individually.
+While this is happening, the server will be periodically sending it `DataAppended` replies acking messages. Note that there can be multiple "Appends Setup" for a given TCP connection. This allows a client to share a connection when producing to multiple Segments.
 
-The EndEventBlock message specifies the size of the append block rather than the BeginAppendBlock message. This means that the size of the data in the block need not be known in advance. This is useful if a client is producing a stream of small messages. It can begin a block, write many messages and then when it comes time to end the block, it can write a partial message followed the EndAppendBlock message, followed by the remaining partial message. This would avoid having headers on all of the messages in the block without having to buffer them in ram in its process.
+A client can optimize its appending by specifying a large value in it's `AppendBlock` message, as the events inside of the block do not need to be processed individually.
+
+The `ApppendBlockEnd` has a `sizeOfWholeEvents` to allow the append block to be less than full. This allows the client to begin writing a block before it has a large number of events. This avoids the need to buffer up events in the client and allows for lower latency.
```

Comment:
This should be move up to the "AppendBlockEnd" section

Reply:
@tkaitchuck  the para ` The `ApppendBlockEnd` has a `sizeOfWholeEvents` to allow the append block to be less than full. This allows the client to begin writing a block before it has a large number of events. This avoids the need to buffer up events in the client and allows for lower latency.` moved inside `AppendBockEnd` section